py/desispec/workflow/procfuncs.py

```diff
@@ -991,6 +1012,136 @@ def joint_fit(ptable, prows, internal_id, queue, reservation, descriptor, z_subm
+
+    return ptable, internal_id
+
+#########################################
+########     Tilenight     ##############
+#########################################
+def submit_tilenight(ptable, prows, calibjobs, internal_id, queue, reservation,
+              dry_run=0, strictly_successful=False, resubmit_partial_complete=True,
+              system_name=None):
+    """
+    Given a set of prows, this generates a processing table row, creates a batch script, and submits the appropriate
+    tilenight job given by descriptor. The returned ptable has all of these rows added to the
+    table given as input.
+
+    Args:
+        ptable, Table. The processing table where each row is a processed job.
+        prows list or array of dicts. Unsubmitted prestdstar jobs that are first steps in tilenight.
+        calibjobs, dict. Dictionary containing 'nightlybias', 'ccdcalib', 'psfnight'
+                       and 'nightlyflat'. Each key corresponds to a Table.Row or
+                       None. The table.Row() values are for the corresponding
+                       calibration job.
+        internal_id, int, the next internal id to be used for assignment (already incremented up from the last used id number used).
+        queue, str. The name of the queue to submit the jobs to. If None is given the current desi_proc default is used.
+        reservation: str. The reservation to submit jobs to. If None, it is not submitted to a reservation.
+        dry_run, int, If nonzero, this is a simulated run. If dry_run=1 the scripts will be written or submitted. If
+                      dry_run=2, the scripts will not be writter or submitted. Logging will remain the same
+                      for testing as though scripts are being submitted. Default is 0 (false).
+        strictly_successful, bool. Whether all jobs require all inputs to have succeeded. For daily processing, this is
+                                   less desirable because e.g. the sciences can run with SVN default calibrations rather
+                                   than failing completely from failed calibrations. Default is False.
+        resubmit_partial_complete, bool. Default is True. Must be used with check_for_outputs=True. If this flag is True,
+                                         jobs with some prior data are pruned using PROCCAMWORD to only process the
+                                         remaining cameras not found to exist.
+        system_name (str): batch system name, e.g. cori-haswell or perlmutter-gpu
+
+    Returns:
+        ptable, Table. The same processing table as input except with added rows for the joint fit job.
+        tnight_prow, dict. Row of a processing table corresponding to the tilenight job.
+        internal_id, int, the next internal id to be used for assignment (already incremented up from the last used id number used).
+    """
+    log = get_logger()
+    if len(prows) < 1:
+        return ptable, None, internal_id
+
```

Comment:
Would it be good to verify that all exposures have the same TILEID?
This is a lower level function but it may be a nice safety mechanism.

Reply:
@akremin verifying that all exposures in a tilenight job have the same TILEID is already taken care of at a lower level in `desi_proc_tilenight` itself; the information about exposures here is only used for setting the time limit.